### PR TITLE
Optimize AnimatedAdapter

### PR DESCRIPTION
**Pull request by goncalossilva**
AnimatedAdapter now works with primitives and mandates that subclasses have unique item ids. This allows for major performance improvements and simplifications.

Empirical testing has shown that small data sets see an average 40% speedup when calculating changes while large data sets see an average 55% speedup (so less than half the time).

### CODE REVIEW

File: Animations/src/main/java/io/doist/recyclerviewext/animations/AnimatedAdapter.java [hunk 1]
```diff
@@ -257,86 +187,153 @@ public synchronized void animateDataSetChanged() {
...
-        private ArrayList2(int capacity) {
-            super(capacity);
+    private static class Items {
+        private long[] mIds;
+        private int[] mChangeHashes;
+        private int mSize;
+
+        public Items() {
+            mIds = new long[0];
+            mChangeHashes = new int[0];
```
Comment (rastislavvasko) [inline Animations/src/main/java/io/doist/recyclerviewext/animations/AnimatedAdapter.java]: We could also call `this(0)` here.

**Comment by rastislavvasko** [comment]
Everything looking good! :+1:

**Comment by goncalossilva** [comment]
Thanks for reviewing!

File: Animations/src/main/java/io/doist/recyclerviewext/animations/AnimatedAdapter.java (hunk 1 above)
Comment (goncalossilva) [inline Animations/src/main/java/io/doist/recyclerviewext/animations/AnimatedAdapter.java]: Woops, had missed this :blush: Improved in 59cca85.

Thanks!

**Pull request merged**